lib/features/notes/presentation/widgets/note_read_button.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:dairy_app/features/notes/presentation/bloc/notes/notes_bloc.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_tts/flutter_tts.dart';
+
+
+class NoteReadIconButton extends StatefulWidget {
+  const NoteReadIconButton({Key? key}) : super(key: key);
+
+  @override
+  State<NoteReadIconButton> createState() => _NoteReadIconButtonState();
+}
+
+class _NoteReadIconButtonState extends State<NoteReadIconButton> {
+  final FlutterTts flutterTts = FlutterTts()
+    ..setLanguage("en-US")
+    ..setSpeechRate(0.5)
+    ..setPitch(1.0);
+  bool isPlaying = false;
+
+  @override
+  void initState() {
+    super.initState();
+    flutterTts.setCompletionHandler(() {
+      setState(() {
```

Comment:
It's not best practice to use `setState(); `
Please use [`ValueNotifier`](https://api.flutter.dev/flutter/foundation/ValueNotifier-class.html) & [`ValueListenableBuilder`](https://api.flutter.dev/flutter/widgets/ValueListenableBuilder-class.html) to achive the same result... Thank you so much!!